src/adapter/src/coord.rs

```diff
@@ -2191,11 +2196,14 @@ impl Coordinator {
         // For compute reconciliation to recognize that an existing dataflow can be reused, we want
         // to advance the `as_of` far enough that it is beyond the `as_of`s of all dataflows that
         // might still be installed on replicas, but ideally not much farther as that would just
-        // increase the wait time until the index becomes readable. We advance the `as_of` to the
-        // meet of the `upper`s of all dependencies, as we know that no replica can have produced
-        // output for that time, so we can assume that no replica `as_of` has been adanced beyond
-        // this time either.
-        let write_frontier = self.least_valid_write(&id_bundle);
+        // increase the wait time until the index becomes readable, and would also prevent warmup
+        // when an index is reading from a REFRESH MV. We advance the `as_of` to
+        // `least_valid_write` - 1. (`least_valid_write` is the meet of the `upper`s of all
+        // dependencies.) This works because we know that no replica could have produced output at
+        // `least_valid_write`, so their write frontiers can't be later. Also, subtracting 1 is
+        // still ok, because we can assume that the compaction window is at least 1.
+        let write_frontier = self.least_valid_write_step_back(&id_bundle);
```

Comment:
We shouldn't call this variable `write_frontier` anymore, because it isn't one. Should we call it `warmup_frontier` instead?

Also, when we calculate `max_compaction_frontier` below, that still needs to be the compaction window applied to the write frontier, not the warmup frontier.

Reply:
This required some non-trivial changes, so I've put this in a separate commit: 63144986c59e0599afb9338655bd3eb954e11a6c
Could you please check?